src/sourcepoint.ts

```diff
@@ -0,0 +1,171 @@
+import { log } from '@guardian/libs';
+import { setCurrentFramework } from './getCurrentFramework';
+import { isGuardianDomain } from './lib/domain';
+import { mark } from './lib/mark';
+import type { Property } from './lib/property';
+import { ACCOUNT_ID, ENDPOINT } from './lib/sourcepointConfig';
+import { invokeCallbacks } from './onConsentChange';
+import { stub } from './stub';
+import type { Framework } from './types';
+
+let resolveWillShowPrivacyMessage: typeof Promise.resolve;
+export const willShowPrivacyMessage = new Promise<boolean>((resolve) => {
+	resolveWillShowPrivacyMessage = resolve as typeof Promise.resolve;
+});
+
+const getProperty = (framework: Framework): Property => {
+	if (framework == 'aus') return 'https://au.theguardian.com';
+	// whichever *.theguardian.com subdomain the page is served on
+	// TODO comment this better
+	return isGuardianDomain() ? null : 'https://test.theguardian.com';
+};
+
+export const init = (framework: Framework, pubData = {}): void => {
+	stub();
+
+	// // make sure nothing else on the page has accidentally
+	// used the _sp_* name as well
+	if (window._sp_) {
+		throw new Error('Sourcepoint global (window._sp_) is already defined!');
+	}
+
+	setCurrentFramework(framework);
+
+	// invoke callbacks before we receive Sourcepoint events
+	invokeCallbacks();
+
+	const frameworkMessageType: string = framework == 'tcfv2' ? 'gdpr' : 'ccpa';
+
+	log('cmp', `framework: ${framework}`);
+	log('cmp', `frameworkMessageType: ${frameworkMessageType}`);
```

Comment:
Do we need all these logs?

Reply:
I've found them pretty useful for debugging  and figured they wouldn't be too noisy since devs can always unsubscribe from cmp logs if they don't need to see them.